src/controls/src/flatmesh.cpp

```diff
@@ -60,13 +62,25 @@ void FlatMesh::setOuterColor(QColor c)
     update();
 }
 
+void FlatMesh::maybeEnableAnimation()
+{
+    if (isVisible() && m_animated) {
+        m_timer.start();
+    } else {
+        m_timer.stop();
+    }
+
+    emit animatedChanged();
```

Comment:
This emit should stay where it was because that's the only case where the value of the animated property actually changes.